content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md

```diff
@@ -0,0 +1,420 @@
+
+```sh
+terraform init
+```
+
+Then run `terraform plan`. This will produce a speculative execution plan that outlines what steps Terraform will take to create the resources defined in the files you set up in the last section.
+
+```sh
+terraform plan
+```
+
+If the plan worked successfully and you're satisfied that it will produce the resources you expect, you can apply it.
+
+```sh
+terraform apply
+```
+
+Before going through with applying the Terraform configuration, this command will prompt you to confirm that you want it to do so. Enter `yes` to apply the configuration.
+
+```
+. . .
+
+Plan: 3 to add, 0 to change, 0 to destroy.
+
+Changes to Outputs:
+  + buildkite-identity = (known after apply)
+
+Do you want to perform these actions?
+  Terraform will perform the actions described above.
+  Only 'yes' will be accepted to approve.
+
+  Enter a value:
+```
+
+After pressing `ENTER`, the command will complete and will output an `buildkite-identity` value.
+
+```
+. . .
+
+Apply complete! Resources: 3 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+buildkite-identity = "<your buildkite identity>"
+```
+
+This is the identity's UIDP (unique identity path), which you configured the `buildkite.tf` file to emit in the previous section. Note this value down, as you'll need it when you test this identity using a Buildkite workflow. If you need to retrieve this UIDP later on, though, you can always run the following `chainctl` command to obtain a list of the UIDPs of all your existing identities.
+
+```sh
+chainctl iam identities ls
+```
+
+You're now ready to edit a Buildkite pipeline in order to test out this identity.
+
+
+## Testing the identity with a Buildkite pipeline
+
+To test the identity you created with Terraform in the previous section, navigate to your Buildkite pipeline. From the Buildkite Dashboard, click **Pipelines** in the top navigation bar and then click on the pipeline you specified in the `buildkite.tf` file. 
+
+From there, click the **Edit Steps** button to add the following commands to a step in your pipeline.
+
+```
+- command: |
+	wget -O chainctl "https://dl.enforce.dev/chainctl/latest/chainctl_linux_$(uname -m)"
+	chmod +x chainctl
+	mv chainctl /usr/bin
+
+	# Assume
+	chainctl auth login --identity-token $(buildkite-agent oidc request-token --audience issuer.enforce.dev) --identity <your buildkite identity>
+```
+	
+
+There are a couple ways you can add commands to an existing Buildkite pipeline, so follow whatever procedure works best for you. 
+
+If you followed the [**Getting Started** guide linked in the prerequisites](), your pipeline will have a structure like this.
+
+```
+steps:
+  - label: "Pipeline upload"
+	command:buildkite-agent pipeline upload
+```
+
+You could add the commands for testing the identity like this.
+
+```
+steps:
+  - label: "Buildkite test"
+	command:
+	- 'buildkite-agent pipeline upload'
+	- 'wget -O chainctl "https://dl.enforce.dev/chainctl/latest/chainctl_linux_$(uname -m)"'
```

Comment:
Depending on the changes to the steps before this, these commands will need to be edited too.